src/decoders/binary_decoder.rs

```diff
@@ -0,0 +1,177 @@
+use crate::checkers::CheckerTypes;
+use crate::decoders::interface::check_string_success;
+
+use super::crack_results::CrackResult;
+use super::interface::Crack;
+use super::interface::Decoder;
+
+use log::{info, trace};
+
+///! Binary Decoder
+pub struct BinaryDecoder;
+
+impl Crack for Decoder<BinaryDecoder> {
+    fn new() -> Decoder<BinaryDecoder> {
+        Decoder {
+            name: "binary",
+            description: "A binary code represents text, computer processor instructions, or any other data using a two-symbol system. The two-symbol system used is often \"0\" and \"1\" from the binary number system. The binary code assigns a pattern of binary digits, also known as bits, to each character, instruction, etc.",
+            link: "https://en.wikipedia.org/wiki/Binary_code",
+            tags: vec!["binary", "base", "decoder"],
+            expected_runtime: 0.01,
+            expected_success: 1.0,
+            failure_runtime: 0.01,
+            normalised_entropy: vec![1.0, 10.0],
+            popularity: 1.0,
+            phantom: std::marker::PhantomData,
+        }
+    }
+
+    /// This function does the actual decoding
+    /// It returns an Option<string> if it was successful
+    /// Else the Option returns nothing and the error is logged in Trace
+    fn crack(&self, text: &str, checker: &CheckerTypes) -> CrackResult {
+        trace!("Trying binary with text {:?}", text);
+        let mut results = CrackResult::new(self, text.to_string());
+        let mut decoded_strings = Vec::new();
+
+        for shift in 1..25 {
+            let decoded_text = binary_to_string(text, shift);
+
+            let decoded_text = decoded_text;
+            decoded_strings.push(decoded_text);
+            let borrowed_decoded_text = &decoded_strings[decoded_strings.len() - 1];
+            if !check_string_success(borrowed_decoded_text, text) {
+                info!(
```

Comment:
```suggestion
                debug!(
```

I think this is debug? I think we'd want to see this at a debug level unless this Is quite rare 🤔

Reply:
Done.